static/app/views/settings/project/projectTeams.tsx

```diff
@@ -1,134 +1,58 @@
-import type {RouteComponentProps} from 'react-router';
-
 import {addErrorMessage} from 'sentry/actionCreators/indicator';
-import {openCreateTeamModal} from 'sentry/actionCreators/modal';
-import {addTeamToProject, removeTeamFromProject} from 'sentry/actionCreators/projects';
+import {
+  useAddTeamToProject,
+  useFetchProjectTeams,
+  useRemoveTeamFromProject,
+} from 'sentry/actionCreators/projects';
 import {hasEveryAccess} from 'sentry/components/acl/access';
+import LoadingError from 'sentry/components/loadingError';
+import LoadingIndicator from 'sentry/components/loadingIndicator';
+import SentryDocumentTitle from 'sentry/components/sentryDocumentTitle';
 import {t, tct} from 'sentry/locale';
 import TeamStore from 'sentry/stores/teamStore';
-import type {Organization, Project, Team} from 'sentry/types';
+import type {Organization, Project} from 'sentry/types';
 import routeTitleGen from 'sentry/utils/routeTitle';
-import DeprecatedAsyncView from 'sentry/views/deprecatedAsyncView';
 import SettingsPageHeader from 'sentry/views/settings/components/settingsPageHeader';
 import TeamSelectForProject from 'sentry/views/settings/components/teamSelect/teamSelectForProject';
 import TextBlock from 'sentry/views/settings/components/text/textBlock';
 import PermissionAlert from 'sentry/views/settings/project/permissionAlert';
 
-type Props = {
+type ProjectTeamsProps = {
   organization: Organization;
   project: Project;
-} & RouteComponentProps<{projectId: string}, {}>;
-
-type State = {
-  projectTeams: null | Team[];
-} & DeprecatedAsyncView['state'];
-
-class ProjectTeams extends DeprecatedAsyncView<Props, State> {
-  getEndpoints(): ReturnType<DeprecatedAsyncView['getEndpoints']> {
-    const {organization, project} = this.props;
-    return [['projectTeams', `/projects/${organization.slug}/${project.slug}/teams/`]];
+};
+
+export function ProjectTeams({organization, project}: ProjectTeamsProps) {
+  const {
+    data: projectTeams,
+    isLoading,
+    isError,
+  } = useFetchProjectTeams({orgSlug: organization.slug, projectSlug: project.slug});
+  const handleAddTeamToProject = useAddTeamToProject({
+    orgSlug: organization.slug,
+    projectSlug: project.slug,
+  });
+  const handleRemoveTeamFromProject = useRemoveTeamFromProject({
```

Comment:
Each one of these hooks call useApi, I think we should pass useApi as an argument. Wdyt?

Reply:
I think I'd rather keep the function signature as simple as possible, and customizing the api client is very rare so I doubt it will be necessary